File: A2Z_sheet/Array/Easy/07_move_zeroes_to_end/solution.ts
function moveZeroes(nums: number[]): void {
    let i: number = 0;
    for(let j = 0; j < nums.length; j++){
        if(nums[j] != 0){
            [nums[i], nums[j]] = [nums[j], nums[i]];
            i++;
        }
    }
};

File: A2Z_sheet/Array/Easy/07_move_zeroes_to_end/solution.rs
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut i = 0;
        for j in 0..nums.len() {
            if nums[j] != 0 {
                if i != j {
                    nums.swap(i, j);
                }
                i += 1;
            }
        }
    }
}
